spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -111,6 +113,37 @@ private void createPythonScript() {
 
   @Override
   public void open() {
+    // try IPySparkInterpreter first
+    iPySparkInterpreter = getIPySparkInterpreter();
```

Comment:
save the value of `property.getProperty("zeppelin.spark.useIPython", "true").equals("true")` instead of calling it twice?

Reply:
opps, this is a bug in the second if block. I should check `InterpreterContext.get() != null`